hw6.py

```diff
@@ -0,0 +1,72 @@
+my_dict = {
+    'Fireman': 101,
+    'Policeman': 102,
+    'Doctor': 103
+}
+
+
+def stats():
+    print(len(my_dict))
+
+
+def add(key, value):
+    if my_dict.get(key) is None:
+        my_dict[key] = value
+        print(f'{key} was successful added to the dictionary')
+    else:
+        print('You are not allowed to edit')
+
+
+def delete(key):
+    if my_dict.get(key) is not None:
+        del my_dict[key]
+        print(f'{key} successfully deleted')
+    else:
+        print(f'There are no {key} in this dictionary')
+
+
+def names():
+    for item in my_dict.keys():
+        print(item)
+
+
+def show(key):
+    if my_dict.get(key) is not None:
+        print(my_dict.get(key))
+    else:
+        print(f'There are no {key} in this dictionary')
+
+
+while True:
+
+    users_input = input('What do you want to do? ')
+
+    match users_input.split()[0]:
```

Comment:
Great work with `match case`. Here is how you can improve your code to use all power of the match case, try it somehow:
```
while True:

    users_input = input('What do you want to do? ')

    match users_input.split():
        case 'stats', *stats_args:
            stats()
        case 'add', name, phone:
            add(name, phone)
        case 'add', *add_args: # invalid command format
            print('Please enter the line in format \'add Name Number\'')
        case 'delete', name:
            delete(users_input.split()[1])
        case 'delete', *delete_args: # invalid command format
            print('Please enter the line in format \'delete Name\'')
        case 'list':
            names()
        case 'show', name: # invalid command format
            show(users_input.split()[1])
        case 'show', *show_args:
            print('Please enter the line in format \'show Name\'')
        case 'exit':
            break
        case _:
            print('I  don\'t understand you. Please use commands: stats, add, delete, list, show or exit')